lib/index.js

```diff
@@ -178,10 +185,10 @@ Socket.prototype._receive = function () {
             return;
         }
     }
-    if (msg == -1) return;
+  if (msg == -1) return '';
```

Comment:
not that subtle when you think about the fact that `null` is an object according to `typeof`, meanwhile `undefined` is its own type.